pkg/surveyutils/jsonschema.go

```diff
@@ -279,15 +299,16 @@ func (o *JSONSchemaOptions) recurse(name string, prefixes []string, requiredFiel
 			return fmt.Errorf("additionalItems is not supported for %s", name)
 			// TODO support additonalItems
 		}
-		err := o.handleArrayProperty(name, t, output, in, out, outErr)
+		err := o.handleArrayProperty(name, t, output, existingValues, askExisting, autoAcceptDefaults, noAsk,
+			ignoreMissingValues, in, out, outErr)
 		if err != nil {
 			return err
 		}
 	case "number":
 		validators := additionalValidators
 		validators = append(validators, FloatValidator())
-		err := o.handleBasicProperty(name, prefixes, numberValidator(required, validators, t), t, output, in, out,
-			outErr)
+		err := o.handleBasicProperty(name, prefixes, numberValidator(required, validators, t), t, output, existingValues, askExisting,
```

Comment:
that's a heck of a parameter list. It seems you also do this at several places. Would it not make sense to introduce a parameter struct?